### PR TITLE
fix(publish): current menu item on the sidebar wont collapse for a published site

### PR DESCRIPTION
**Pull request by Harshita-mindfire**
This PR aims the fix the issue with the current item on the sidebar not collapsing. 

# Dendron Extended PR Checklist

- NOTE: the links don't work. you'll need to go into the wiki and use lookup to find the note until we fix some issues in the markdown export

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
  - [x] check whether code be simplified
  - [x] check if similar function already exist in the codebase. if so, can it be re-used?
  - [x] check if this change adversely impact performance
- Operations
  - [x] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [x] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics
- [~] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended
- [~] can we track the performance of this change to know if it is _successful_?
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

### Basics

- [~] [Write Tests](dev.process.qa.test) 
- [ ] [Confirm existing tests pass](dev.process.qa.test)
- [x] [Confirm manual testing](dev.process.qa.test) 
- [ ] Common cases tested
- [ ] 1-2 Edge cases tested
- [ ] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

### Extended
NA
## Docs
NA

### Extended
NA

### CODE REVIEW

**Review by kevinslin — CHANGES_REQUESTED** [review]
looks good, one issue when trying to export is that `DendronTreeMenu` can return undefined which is not a valid react construct. 

```
$ next build && next export
info  - Loaded env from /Users/kevinlin/code/dendron/packages/nextjs-template/.env.production
Failed to compile.

./components/DendronLayout.tsx:49:8
Type error: 'DendronTreeMenu' cannot be used as a JSX component.
  Its return type '"" | Element | null | undefined' is not a valid JSX element.
    Type 'undefined' is not assignable to type 'Element | null'.

  47 |         </div>
  48 |       )}
> 49 |       <DendronTreeMenu
     |        ^
  50 |         {...props}
  51 |         collapsed={isCollapsed && isResponsive}
  52 |         setCollapsed={setCollapsed}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
kevinlin@Kevins-MacBook-Pro-2 nextjs-template % yarn export
```

File: packages/nextjs-template/components/DendronTreeMenu.tsx [hunk 1]
```diff
@@ -99,15 +98,17 @@ export default function DendronTreeMenu(
   };
 
   return (
```
Comment (kevinslin) [inline packages/nextjs-template/components/DendronTreeMenu.tsx:100]: this should return `<MenuView .../> || null` if `noteActiveId` isn't defined

**Review by kevinslin — APPROVED** [review]
